tests/test_nmf_mfcc.py

```diff
@@ -0,0 +1,152 @@
+        assert nmf_mfcc.random_seed == None
+
+        # No random_state, but random_seed initialized
+        nmf_mfcc = nussl.NMF_MFCC(signal, num_sources=2, num_templates=6, num_iterations=5, random_seed=0)
+        assert nmf_mfcc.clusterer.random_state == nmf_mfcc.random_seed == 0
+
+    def test_2_sin_waves(self):
+        sr = nussl.constants.DEFAULT_SAMPLE_RATE
+        dur = 3  # seconds
+        length = dur * sr
+        sine_wave_1 = np.sin(np.linspace(0, 440 * 2 * np.pi, length))
+        sine_wave_2 = np.sin(np.linspace(0, 440 * 5 * 2 * np.pi, length))
+        signal = sine_wave_1 + sine_wave_2
+        test_audio = nussl.AudioSignal()
+        test_audio.load_audio_from_array(signal)
+
+        # Set up NMMF MFCC
+        nmf_mfcc = nussl.NMF_MFCC(test_audio, num_sources=2, num_templates=2, num_iterations=15)
+
+        # and run
+        nmf_mfcc.run()
+        sources = nmf_mfcc.make_audio_signals()
+
+        for i, source in enumerate(sources):
+            output_file_name = str(i) + '.wav'
+            source.write_audio_to_file(output_file_name)
+
+    def test_piano_and_synth(self):
+        # Load input file
+        input_file_name = os.path.join('..', 'input', 'piano_and_synth_arp_chord_mono.wav')
+        signal = nussl.AudioSignal(path_to_input_file=input_file_name)
+
+        # Set up NMMF MFCC
+        nmf_mfcc = nussl.NMF_MFCC(signal, num_sources=2, num_templates=6, num_iterations=50, random_seed=0)
+        # and run
+        nmf_mfcc.run()
+        sources = nmf_mfcc.make_audio_signals()
+        for i, source in enumerate(sources):
+            output_file_name = str(i) + '.wav'
+            source.write_audio_to_file(output_file_name)
+
+    def test_piano_and_synth_stereo(self):
+        # Load input file
+        input_file_name = os.path.join('..', 'input', 'piano_and_synth_arp_chord_stereo.wav')
+        signal = nussl.AudioSignal(path_to_input_file=input_file_name)
+
+        # Set up NMMF MFCC
+        nmf_mfcc = nussl.NMF_MFCC(signal, num_sources=2, num_templates=6, num_iterations=150, random_seed=0)
+        # and run
+        nmf_mfcc.run()
+        sources = nmf_mfcc.make_audio_signals()
+        for i, source in enumerate(sources):
+            output_file_name = str(i) + '.wav'
+            source.write_audio_to_file(output_file_name)
+
+    def test_piano_and_synth_stereo_to_mono(self):
+        # Load input file
+        input_file_name = os.path.join('..', 'input', 'piano_and_synth_arp_chord_stereo.wav')
+        signal = nussl.AudioSignal(path_to_input_file=input_file_name)
+
+        # Set up NMMF MFCC and convert input audio signal to mono
+        nmf_mfcc = nussl.NMF_MFCC(signal, num_sources=2, num_templates=6, distance_measure="euclidean",
+                                  num_iterations=15, random_seed=0, convert_to_mono=True)
+        # and run
+        nmf_mfcc.run()
+        sources = nmf_mfcc.make_audio_signals()
+        for i, source in enumerate(sources):
```

Comment:
Ditto. Especially, because when you run all of these tests in succession (which is the most common way they'll get run, by triggering all of the tests to run), each of these test makes the same filenames and will get overwritten 3-4 times, rendering all but the last run useless. And no one will listen to these files regularly, which is why you have the benchmark test below, to make sure they're numerically equivalent.